src/core/dbus-execute.c

```diff
@@ -3868,6 +3888,50 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (streq(name, "CgroupNFTSet")) {
+                int family;
+                const char *table, *set, *family_str;
+
+                r = sd_bus_message_read(message, "(iss)", &family, &table, &set);
+                if (r < 0)
+                        return r;
+
+                if (family > 0) {
+                        family_str = nfproto_to_string(family);
+                        if (!family_str)
+                                return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Cgroup NFT family not valid");
+                }
+
+                if (isempty(table))
+                        return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Cgroup NFT table name not valid");
+
+                if (isempty(set))
+                        return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Cgroup NFT set name not valid");
+
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+
+                        if (family == 0) {
+                                c->cgroup_nft_family = 0;
+                                c->cgroup_nft_table = mfree(c->cgroup_nft_table);
+                                c->cgroup_nft_set = mfree(c->cgroup_nft_set);
+                                unit_write_settingf(u, flags, name, "%s=", name);
+                        } else {
+                                c->cgroup_nft_family = family;
+
+                                r = free_and_strdup(&c->cgroup_nft_table, table);
+                                if (r < 0)
+                                        return r;
+
+                                r = free_and_strdup(&c->cgroup_nft_set, set);
+                                if (r < 0)
+                                        return r;
+
+                                unit_write_settingf(u, flags, name, "%s=%s:%s:%s", name, family_str, table, set);
```

Comment:
this of course will fall apart if people include newlines in the strings. So at the very least strng_is_safe() should be used.

But then names that include `:` will still break the logic here. Hence, mabye use cescape() and force them to be escaped.